app/packages/dataset/src/Dataset.tsx

```diff
@@ -55,20 +55,24 @@ export function Dataset({ datasetName, environment }) {
     flex-direction: column;
     min-width: 660px;
   `;
+  const themePalette = theme?.palette;
+  const ThemeWrapper = themePalette ? ThemeContext.Provider : Fragment;
 
   return (
-    <Container>
-      <Suspense fallback={loadingElement}>
-        <DatasetLoader
-          datasetQueryRef={datasetQueryRef}
-          initialState={initialState}
-        >
-          <ViewBar />
-          <CoreDataset />
-        </DatasetLoader>
-      </Suspense>
-      <div id="modal" />
-    </Container>
+    <ThemeWrapper value={themePalette}>
```

Comment:
When this is `<Fragment  value={false} />` won't it throw?

Maybe...

```ts
  const themePalette = theme?.palette;
  const ThemeWrapper = themePalette ? ThemeContext.Provider : Fragment;
  const themeProps = themePalette ? {value: themePalette} : {};
  return (
    <ThemeWrapper {...themeProps}>
```

Reply:
Good idea! It didn't throw when I omitted theme prop but it's better to do this.